precompiles/vesting/tx.go

```diff
@@ -74,8 +76,8 @@ func (p Precompile) FundVestingAccount(
 		return nil, err
 	}
 
-	// Check if the origin matches the funders address
-	if origin != funderAddr {
+	// if caller address is origin, check if the funder matches the origin
```

Comment:
I think that it's clear from line 80 what this part of the code is checking. Would be more informative to state in the comment why we are checking this.